### ISSUE TITLE
Change scattering_matrix to broadening_matrix?

### ISSUE THREAD

**Issue by zerothi** [zerothi/sisl#529]
**Describe the issue**

the name `scattering_matrix` is misleading due to the connection with `S`-matrix terminology.

I think we should rename the matrices accordingly.

Suggestions:
- `broadening_matrix`
- others?

@sofiasanz @tfrederiksen inputs are most welcome here!

**Comment by sofiasanz** [comment]
Agreed! For me `scattering_matrix` was also a bit confusing to refer to it. 
I think `broadening_matrix` is a good name for this quantity (as far as I know I have always seen it called by this name).

**Comment by tfrederiksen** [comment]
I agree, `broadening_matrix` is better (more conventional).

**Comment by zerothi** [comment]
Thanks, I'll adjust accordingly! Please suggest if other things come to mind!

**Comment by zerothi** [comment]
Fixed!
